src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+     * @param transcriptId The ID of the transcript to get from the FASTA file.
+     * @param transcriptIdMap A map from transcriptId to MappedTranscriptIdInfo, which tells us how to pull information for the given {@code transcriptId} out of the given {@code transcriptFastaReferenceDataSource}.
+     * @param transcriptFastaReferenceDataSource A {@link ReferenceDataSource} for the GENCODE transcript FASTA file.
+     * @return The coding sequence for the given {@code transcriptId} as represented in the GENCODE transcript FASTA file.
+     */
+    private static String getCodingSequenceFromTranscriptFasta( final String transcriptId,
+                                                                final Map<String, MappedTranscriptIdInfo> transcriptIdMap,
+                                                                final ReferenceDataSource transcriptFastaReferenceDataSource) {
+
+        final MappedTranscriptIdInfo transcriptMapIdAndMetadata = transcriptIdMap.get(transcriptId);
+
+        if ( transcriptMapIdAndMetadata == null ) {
+            throw new UserException.BadInput( "Unable to find the given Transcript ID in our transcript list (not in given transcript FASTA file): " + transcriptId );
+        }
+
+        final SimpleInterval transcriptInterval = new SimpleInterval(
+                transcriptMapIdAndMetadata.mapKey,
+                transcriptMapIdAndMetadata.codingSequenceStart,
+                transcriptMapIdAndMetadata.codingSequenceEnd
+        );
+
+        return transcriptFastaReferenceDataSource.queryAndPrefetch( transcriptInterval ).getBaseString();
+    }
+
+    /**
+     * Returns whether a variant is in a coding region based on its primary and secondary {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification}.
+     * @param varClass Primary {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification} of a variant.
+     * @param secondaryVarClass Secondary {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification} of a variant.
+     * @return {@code true} if the corresponding variant is in a coding region; {@code false} otherwise.
+     */
+    @VisibleForTesting
+    static boolean isVariantInCodingRegion(final GencodeFuncotation.VariantClassification varClass,
+                                           final GencodeFuncotation.VariantClassification secondaryVarClass ) {
+
+        Utils.nonNull( varClass );
+
+        if ( varClass == GencodeFuncotation.VariantClassification.SPLICE_SITE ) {
+            Utils.nonNull( secondaryVarClass );
+            return secondaryVarClass != GencodeFuncotation.VariantClassification.INTRON;
+        }
+        else {
+            return  (varClass == GencodeFuncotation.VariantClassification.MISSENSE)        ||
```

Comment:
You should make a set ... something like
```
final static Set<VariantClassificiation> codingVariantClassifications = Sets.newHashSet(GencodeFuncotation.VariantClassification.MISSENSE, ... , GencodeFuncotation.VariantClassification.START_CODON_DEL);

......snip....
return codingVariantClassifications.contains(varClass);
```

Reply:
Sounds good.  Fixed.